local/compose/run.go

```diff
@@ -0,0 +1,141 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compose
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/compose-spec/compose-go/types"
+	"github.com/docker/compose-cli/api/compose"
+	convert "github.com/docker/compose-cli/local/moby"
+	apitypes "github.com/docker/docker/api/types"
+	moby "github.com/docker/docker/pkg/stringid"
+)
+
+func (s *composeService) CreateOneOffContainer(ctx context.Context, project *types.Project, opts compose.RunOptions) (string, error) {
+	service, err := project.GetService(opts.Name)
+	if err != nil {
+		return "", err
+	}
+	if err = s.ensureImagesExists(ctx, project); err != nil {
+		return "", err
+	}
+	if err := s.ensureProjectNetworks(ctx, project); err != nil {
+		return "", err
+	}
+	if err := s.ensureProjectVolumes(ctx, project); err != nil {
+		return "", err
+	}
+	if err = s.ensureRequiredServices(ctx, project, service); err != nil {
+		return "", err
+	}
```

Comment:
AFAICT this is equivalent to `s.Create(filteredProject)` Where `filteredProject` is a subset of the original project to only contains run target (transitive) dependencies

Reply:
I'll look into this, to reuse `s.Create(filteredProject)` I guess we only need to remove the required service to we only start dependencies at this stage